### PR TITLE
[e2e ingress-gce] Fix race condition for appending services and ingresses

### PR DESCRIPTION
**Pull request by MrHohn**
**What this PR does / why we need it**:
From https://github.com/kubernetes/ingress-gce/issues/219, how the test appends services and ingresses is not thread-safe. This PR fixes it by using channel.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #https://github.com/kubernetes/ingress-gce/issues/219

**Special notes for your reviewer**:
/assign @nicksardo 
I picked this up as it is fun :) Hoping you haven't worked on it yet.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nicksardo** [comment]
I'm also confused by this shared reference to `currentNum`. What is the purpose of it?

**Comment by MrHohn** [comment]
> I'm also confused by this shared reference to currentNum. What is the purpose of it?

Sorry, that was implemented in a very confusing way. The reference to `currentNum` is not shared among goroutines, it is only used serially. The idea is to generate a suffix for a goroutine before it starts.

**Comment by MrHohn** [comment]
Actually it can just be an integer I think...

File: test/e2e/network/scale/ingress.go [hunk 1]
```diff
@@ -180,24 +180,26 @@ func (f *IngressScaleFramework) RunScaleTest() []error {
 	}
 
 	// currentNum keeps track of how many ingresses have been created.
-	currentNum := new(int)
+	currentNum := 0
```
Comment (MrHohn) [inline test/e2e/network/scale/ingress.go]: > Actually it can just be an integer I think...

Changed to an integer.

**Pull request merged**